server/app/mutations/grid_services/delete.rb

```diff
@@ -15,14 +15,29 @@ def validate
 
     def execute
       nodes = self.grid_service.grid_service_instances.map{ |i| i.host_node }
+      notify_lb_remove if self.grid_service.linked_to_load_balancer?
       self.grid_service.destroy
       nodes.each do |node|
         notify_node(node) if node
       end
     end
 
+    # @param [HostNode] node
     def notify_node(node)
       RpcClient.new(node.node_id).notify('/service_pods/notify_update', 'terminate')
     end
+
+    def notify_lb_remove
```

Comment:
Actually, this could be fixed, if the agent `/load_balancers/remove_service` RPC fails with an error, then it will propagate here and cause the `self.grid_service.destroy` to be skipped... the service delete will fail, and it can be retried later by the client... :)

Unfortunately it looks like the agent `Celluloid::Actor[:lb_configurer].remove_config` will just swallow exceptions, so it will currently fail silently on things like etcd errors :(